zeppelin-server/src/main/java/org/apache/zeppelin/realm/jwt/KnoxJwtRealm.java

```diff
@@ -182,6 +193,33 @@ protected boolean validateSignature(SignedJWT jwtToken) {
     return valid;
   }
 
+  /**
+   * Validate that the expiration time of the JWT token has not been violated.
+   * If it has then throw an AuthenticationException. Override this method in
+   * subclasses in order to customize the expiration validation behavior.
+   *
+   * @param jwtToken
+   *            the token that contains the expiration date to validate
+   * @return valid true if the token has not expired; false otherwise
+   */
+  protected boolean validateExpiration(SignedJWT jwtToken) {
+    boolean valid = false;
+    try {
+      Date expires = jwtToken.getJWTClaimsSet().getExpirationTime();
```

Comment:
Does LOCALE matter here ?

Reply:
It's taken from https://github.com/apache/ranger/blob/master/security-admin/src/main/java/org/apache/ranger/security/web/filter/RangerSSOAuthenticationFilter.java#L467-L491.
ExpirationTime is passed as an epoch, so it should not matter.

Reply:
Yes, that I understood LOCALE for this context is time zone, yes it should not matter.

Reply:
Thanks, LGTM